modules/manifold/src/utils/utils.js

```diff
@@ -388,6 +358,32 @@ export function zipObjects(arrays, joinField, rename) {
   });
 }
 
+export function product(collectionArr) {
+  let result = [];
+  function recur(collection) {
+    if (collection.length < collectionArr.length) {
+      collectionArr[collection.length].forEach(value => {
+        recur(collection.concat(value));
+      });
+    } else {
+      result.push(collection);
+    }
+  }
+  recur([]);
+  return result;
+}
+
+// todo: to be consolidated with `computeNumericalFeatureDomain`
+export function getColumnDomain(values) {
```

Comment:
From the code above it seems domain has more values than [min, max].
```
domain[domain.length - 1]
```

Are we using the term `domain` for different purposes? 
If yes shall we change this function to `GetColumnMinMax` instead? (and it's derived variables)